### ISSUE TITLE
[FEATURE] Implement pytorch dataset for healpix grid from ERA5

### ISSUE THREAD

**Issue by ankitpatnala** [clessig/atmorep#79]
**Is your feature request related to a problem? Please describe.**
Right now, ERA5 is on a rectangular grid which poses a problem of singularity on poles. Healpix grid [healpix_doc](https://healpix.sourceforge.io/) is an alternative way of gridding type which is designed for spherical coordinates such as the Earth.

**Describe the solution you'd like**
We want to continue stepwise
-  [ ] Understand how to navigate through healpix grid. Access the geolocation using parameters of healpix grid.
-  [ ] Conversion of rectangular grid to healpix grid using CDO
-  [ ] Tokenization of healpix grid
-  [ ] Development of healpix PyTorch dataset

**Comment by SavvasMel** [comment]
Hello,

The current branch that we created for HEALPix code development is based on the `main` branch. However, we set up everything under the `HClimrep_hackathon_2024` so that AtmoRep can run smoothly (at least for me and Erik if I remember correctly).

Anyway, I did the necessary changes to the branch dedicated to HEALPix so that it can resemble the `HClimrep_hackathon_2024` one. Do you think that it would be ok to push these changes so that we can all start from the same version?

**Comment by clessig** [comment]
You mean healpix for the input data?

**Comment by SavvasMel** [comment]
Yes exactly! I hope my comment above was not confusing. My point was that under the `HClimrep_hackathon_2024`, both training and evaluation run smoothly for me. 

So I was aiming for the branch `79-melidonis-feature-implement-pytorch-dataset-for-healpix-grid-from-era5` to be mainly a copy of the branch `HClimrep_hackathon_2024` and then take it from there. I have already made the necessary changes locally.

**Comment by ankitpatnala** [comment]
@SavvasMel The hackathon branch was meant for hackathon and the necessary files are under folder jsc_scripts. it may not be updated.  I would suggest it should be copy of develop branch rather main branch and then you can checkout the folder jsc_scripts only from the hackathon branch.

**Comment by SavvasMel** [comment]
Ok then I guess `79-melidonis-feature-implement-pytorch-dataset-for-healpix-grid-from-era5` will stay as copy of the `main`. I created another remote branch from `develop` under the name `82-melidonis-feature-implement-pytorch-dataset-for-healpix-grid-from-era5`.

**Comment by clessig** [comment]
Hi Savvas, can you please use the convention for naming branches. Also, what do you want to implement?

**Comment by SavvasMel** [comment]
Yes of course, my apologies.

I am not trying to implement something right now or in the very near future since things are in an early stage. It is that  a branch related to the HEALPix research direction has been created  (`79-melidonis-feature-implement-pytorch-dataset-for-healpix-grid-from-era5`) and this branch is copy of `main`. I thought that `HClimrep_hackathon_2024` includes the most recent updates over main (possibly unmerged updates from other branches too). I would like to figure out what (updated) version of the code I/we should have as a base for the new research direction. This direction will mainly affect the dataloaders and the tokenization parts.
